pkgs/os-specific/darwin/apple-sdk-12.3/libSystem.nix

```
@@ -0,0 +1,99 @@
+{
+  buildPackages,
+  darwin-stubs,
+  stdenvNoCC,
+}:
+
+stdenvNoCC.mkDerivation {
```

```suggestion
runCommand "libSystem-version" { nativeBuildInputs = [ rewrite-tbd ]; } ''
```

can this just be a runCommandLocal?

---

It’s copying the files not linking them. Using `runCommandLocal` would require downloading the full set of stubs just to build Libsystem, which is 527 MiB versus just 24 MiB for Libsystem.

---

i see. so the sum of all of the derivations is less than the stubs then?

---

i guess they are coping the files rather than just symlinking, so i guess makes less sense.

---

> i see. so the sum of all of the derivations is less than the stubs then?

The way the SDK is currently set up, packages only download the frameworks they need.